### PR TITLE
filter out cross-network non mTLS lb eps

### PR DESCRIPTION
**Pull request by stevenctl**
When a sidecared pod in network1 tries to reach a non-sidecared pod in network2, it will result in `TLS error: 268435703:SSL routines:OPENSSL_internal:WRONG_VERSION_NUMBER`.

This is problematic in a scenario where the non-sidecared service exists in both networks. We would loadbalance between endpoints we can reach and ones we cannot. It may be possible to do TLS termination for these endpoints at the gateway, but for now we should at least remove the unreachable endpoints. 

* Don't build cross-network endpoints with tlsmode disabled. 
* Test that cross network VMs still work in integration tests
* Test that we hit all network-local non-sidecar pods in integration tests (naked service)
* Update XDS tests to set tls mode properly

pilot-multicluster test runs: [95](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/26486/integ-pilot-multicluster-tests_istio/95) [97](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/26486/integ-pilot-multicluster-tests_istio/97). 

Fixes https://github.com/istio/istio/issues/26517

cc @costinm @howardjohn @hzxuzhonghu 

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: pkg/test/framework/resource/cluster.go [hunk 1]
```diff
@@ -58,6 +58,15 @@ func (c Clusters) Names() []string {
 	return names
 }
 
+// ByNetwork returns a map of network name to a subset of clusters
+func (c Clusters) ByNetwork() map[string]Clusters {
```
Comment (stevenctl) [inline pkg/test/framework/resource/cluster.go:62]: We should try to fold the accessors for control plane and network topologies into `Cluseters` and get rid of the methods on Environment.

File: pilot/pkg/xds/xds_test.go [hunk 2]
```diff
@@ -390,6 +405,7 @@ spec:
   location: MESH_INTERNAL
   endpoints:
   - address: 10.10.10.30
+    serviceAccount: svc-acc
```
Comment (stevenctl) [inline pilot/pkg/xds/xds_test.go:394]: Tells us that the service instance will use mTLS.

File: releasenotes/notes/26486.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+apiVersion: release-notes/v2
+kind: feature
+area: traffic-management
+issue:
+  - 26517
+releaseNotes:
+  - |
+    **Fixed** Remove bad endpoints for non-injected workloads across networks.
```
Comment (howardjohn) [inline releasenotes/notes/26486.yaml:8]: "bad" seems a bit vague. Could say unreachable or something maybe

**Pull request merged**